src/main/java/com/iota/iri/service/API.java

```diff
@@ -531,8 +531,16 @@ public static void incEllapsedTime_getTxToApprove(long ellapsedTime) {
         Hash referenceHash = null;
         if(reference != null) {
             referenceHash = new Hash(reference);
-            if(!TransactionViewModel.exists(instance.tangle, referenceHash)) {
+            if (!TransactionViewModel.exists(instance.tangle, referenceHash)) {
+                //reference not found
                 referenceHash = null;
+            } else {
+                TransactionViewModel transactionViewModel = TransactionViewModel.fromHash(instance.tangle, referenceHash);
+                if (transactionViewModel.snapshotIndex() != 0
```

Comment:
Does this != 0 check mean that a very large block of unconfirmed transactions can be used as a reference? Is that safe?